packages/gatsby-dev-cli/src/watch.js

```diff
@@ -17,7 +17,7 @@ function watch(root, packages, { scanOnce, quiet }) {
     const ignoreRegs = [
       /[\/\\]node_modules[\/\\]/i,
       /\.git/i,
-      new RegExp(`${p}[\\/]src[\\/]`, `i`),
+      new RegExp(`${p}[\\/\\\\]src[\\/\\\\]`, `i`),
```

Comment:
yes your escape is right, but it doesn't check for paths with `\windows\paths\use\these` so it's checking an additional `\` character, which in a regex is `\\` to escape it and in a string `\\\\` -> `\\`